spec/features/fill_in_additional_information_page_spec.rb

```diff
@@ -19,15 +19,21 @@
     expect(confirmation_of_supplied_details_page).to be_displayed
   end
 
-  scenario "incorrectly will provide errors" do
-    pending("while we wait for a test environment to understand how the s3 bucket timings will work")
-    fail
-  end
-
   scenario "without uploading a file will not display 'Remove file' link on CoSD page" do
     additional_information_page.load
     additional_information_page.next
 
     expect(confirmation_of_supplied_details_page.confirmation_of_additional_information_answers).not_to have_link(t('components.confirmation_of_supplied_details.remove_file_link'), href: remove_rtf_path)
   end
+
+  scenario "correctly will leave a file in the s3 bucket" do
+    additional_information_page.load
+
+    given_i_am(:company01)
+
+    answer_upload_additional_information_question
+    additional_information_page.next
+
+    expect(keys_in_bucket).to include Store.last.hash_store[:additional_information_answers][:upload_additional_information]
```

Comment:
You are testing against internals of the application here.  Should you not be testing that the API was sent the correct key and that it is in the bucket ?  Noone cares about Store.last.hash_store really ?

Reply:
That test is in the stubbed API submission in the `fill_in_whole_form` spec. This is a weird hybrid unit test thing. Perhaps @sivvoy will have a better solution?